javascript/ql/src/Security/CWE-693/InsecureHelmet.ql

```diff
@@ -0,0 +1,58 @@
+/**
+ * @name Insecure configuration of Helmet security middleware
+ * @description The Helmet middleware is used to set security-related HTTP headers in Express applications. This query finds instances where the middleware is configured with important security features disabled.
+ * @kind problem
+ * @problem.severity error
+ * @security-severity 5.0
+ * @precision high
+ * @id javascript/insecure-helmet-configuration
+ * @tags security
+ *        cwe-693
+ *        cwe-1021
+ */
+
+import semmle.javascript.frameworks.ExpressModules
+
+class HelmetProperty extends Property {
+  ExpressLibraries::HelmetRouteHandler helmet;
+
+  HelmetProperty() {
+    helmet.(DataFlow::CallNode).getAnArgument().asExpr().(ObjectExpr).getAProperty() = this
+  }
+
+  ExpressLibraries::HelmetRouteHandler getHelmet() { result = helmet }
+
+  predicate isFalse() { this.getInit().(BooleanLiteral).getBoolValue() = false }
+
+  predicate isImportantSecuritySetting() {
+    this.getName() in ["frameguard", "contentSecurityPolicy"]
+    or
+    // read from data extensions to allow enforcing other settings
+    requiredHelmetSecuritySetting(this.getName())
+  }
+}
+
+/*
+ * Extend the required Helmet security settings using data extensions.
+ * Docs: https://codeql.github.com/docs/codeql-language-guides/customizing-library-models-for-javascript/
+ * For example:
+ *
+ * extensions:
+ *  - addsTo:
+ *      pack: codeql/javascript-all
+ *      extensible: requiredHelmetSecuritySetting
+ *    data:
+ *      - name: "frameguard"
+ *
+ * Note: `frameguard` is an example: the query already enforces this setting, so it is not necessary to add it to the data extension.
+ */
+
+extensible predicate requiredHelmetSecuritySetting(string name);
```

Comment:
Both on CI and on my local machine I'm getting this error: 
```
The query depends on an extensional predicate requiredHelmetSecuritySetting which has not been defined.
```

I have not worked with `extensible` predicates, so I'm not sure what the fix could be, but maybe you have to add an empty extension, just so there is some data?

Reply:
Yep, good point. Sorry to waste your time by not checking.

I'd assumed if it didn't find a matching extension file it'd just carry on regardless. Looks like it needs at least one matching model; and I forgot to even test that assumption locally.

What I've done now is defined a model file in the `javascript/ql/lib/semmle/frameworks` directory, which is used to define the required security settings.

I've also added a `README.md` to help guide users on how to customize this query with a data extension of their own.